secant/Features/OnboardingFlow/Views/OnboardingContentView.swift

```diff
@@ -15,71 +15,54 @@ struct OnboardingContentView: View {
-                                }
-                            )
-                        )
-                        .offset(y: viewStore.offset - height / circularFrameOffsetCoeffcient)
-                        .transition(.scale(scale: 2).combined(with: .opacity))
+            let scale = imageScale
+            let imageWidth: CGFloat = width * scale
+            let imageXOffset: CGFloat = (width - imageWidth) / 2
+
+            let image = viewStore.steps[viewStore.index].background
+                .resizable()
+                .aspectRatio(contentMode: .fit)
+                .frame(width: imageWidth)
+                .offset(x: imageXOffset)
+
+            let title = Text(viewStore.steps[viewStore.index].title)
+                .titleText()
+                .lineLimit(0)
+                .minimumScaleFactor(0.1)
+                .padding(EdgeInsets(top: 0, leading: 10, bottom: 5, trailing: 10))
+
+            let text = Text(viewStore.steps[viewStore.index].description)
+                .paragraphText()
+                .lineSpacing(2)
+                .padding(EdgeInsets(top: 0, leading: 10, bottom: 10, trailing: 10))
+
+            if viewStore.isInitialStep {
+                VStack(alignment: .leading) {
+                    image
+                    title
+                    text
+                    Spacer()
                 }
-            }
-            ZStack {
-                ForEach(0..<viewStore.steps.count, id: \.self) { stepIndex in
-                    VStack(spacing: viewStore.isFinalStep ? 50 : 15) {
-                        HStack {
-                            Text(viewStore.steps[stepIndex].title)
-                                .titleText()
-                                .lineLimit(0)
-                                .minimumScaleFactor(0.1)
-                            if !viewStore.isFinalStep {
-                                Spacer()
-                            }
-                        }
-                        
-                        Text(viewStore.steps[stepIndex].description)
-                            .paragraphText()
-                            .lineSpacing(2)
-                            .opacity(0.53)
-                    }
-                    .opacity(stepIndex == viewStore.index ? 1: 0)
-                    .padding(.horizontal, 35)
-                    .frame(width: width, height: height)
+                .ignoresSafeArea()
+            } else if viewStore.isFinalStep {
+                VStack(alignment: .leading) {
+                    title
+                        .padding(.top, 73 * imageScale)
+                    text
+                    image
+                    Spacer()
                 }
+            } else {
+                VStack(alignment: .leading) {
+                    image
+                        .padding(.top, 73 * imageScale)
+
+                    title
+                    text
+                    Spacer()
+                }
+                .ignoresSafeArea()
```

Comment:
I think this should be fine. Because content in this stack will always be above any button that is in  bottom of the screen. But to be sure I changed a code little.

This is the only way how to be in compliant with designs. It's super easy to update scale of that image and if it should ignore safe area or not. So we can tweak it when it comes to finalisation of the UI.